src/stores/AppStore.ts

```diff
@@ -1057,11 +1063,11 @@ export class AppStore {
     };
 
     private recalculateStatsRequirements() {
-        if (!this.activeFrame) {
+        if (!this.requiredFrame) {
             return;
         }
 
-        const updatedRequirements = RegionWidgetStore.CalculateRequirementsArray(this.activeFrame, this.widgetsStore.statsWidgets);
+        const updatedRequirements = RegionWidgetStore.CalculateRequirementsArray(this.requiredFrame, this.widgetsStore.statsWidgets);
```

Comment:
this is where the complexity lies with this feature. Previously, we assumed that the only changes to requirements were for the active frame. But now, when we change the effective frame, this will result in _two_ images changing their requirements, won't we?